### PR TITLE
Enforce wordle rules and color letters in UI accordingly

### PR DESCRIPTION
**Pull request by Z1Epime**
- Added determination of the letter color for the UI + tests
- Changed input type from 1 TextBox per word to 1 TextBox per letter
- Letters in the UI are colored according to existence in the solution word

### CODE REVIEW

**Review by MibFromGit — APPROVED** [review]

File: DataAccess/Model/Word.cs [hunk 1]
```diff
@@ -8,6 +8,8 @@ namespace DataAccess.Model
 {
     public class Word
     {
+        public const int WordLength = 5;
```
Comment (MibFromGit) [inline DataAccess/Model/Word.cs:11]: Das wurde schon gemerged

File: esWordle.Tests/RuleEnforcerTests.cs [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+                yield return new TestCaseData("Y", 1, new Word("Knast"));
+                yield return new TestCaseData("Y", 2, new Word("Knast"));
+                yield return new TestCaseData("Y", 3, new Word("Knast"));
+                yield return new TestCaseData("Y", 4, new Word("Knast"));
+
+                yield return new TestCaseData("Z", 0, new Word("Knast"));
+                yield return new TestCaseData("Z", 1, new Word("Knast"));
+                yield return new TestCaseData("Z", 2, new Word("Knast"));
+                yield return new TestCaseData("Z", 3, new Word("Knast"));
+                yield return new TestCaseData("Z", 4, new Word("Knast"));
```
Comment (SirPatagon) [inline esWordle.Tests/RuleEnforcerTests.cs]:
```suggestion
                var word = new Word("Knast");
                var alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ".Where(a => !word.Letters.ToUpperInvariant().Contains(a)); // Get alphabet without any letters of the word
                
                foreach (var letter in alphabet)
                {
                    for (int i = 0; i < 5; i++)
                        yield return new TestCaseData(letter.ToString(), i, wort);
                }
```

Comment (Z1Epime) [reply esWordle.Tests/RuleEnforcerTests.cs]: Applied

**Comment by Z1Epime** [comment]
Looks good 2 me 👍

**Pull request merged**